dali/base/dasds.cpp

```diff
@@ -5760,45 +5761,81 @@ class CStoreHelper : implements IStoreHelper, public CInterface
-                iXmlFileTmpStore->remove();
-                return;
-            }
 
             // Save store as binary
             OwnedIFile iBinaryFileTmpStore;
+            constexpr const char *temporaryBinaryFileSaveName = "dali_store_tmp.bin";
+            OwnedIFileIO iBinaryFileIOTmpStore = createUniqueFile(location, temporaryBinaryFileSaveName, NULL, tmpStoreName);
+            iBinaryFileTmpStore.setown(createIFile(tmpStoreName));
+
+            // Execute save operations
+            unsigned xmlCrc{0};
             unsigned binaryCrc{0};
-            unsigned *binaryCrcPtr{nullptr};
-            if (saveBinary)
+            bool binaryStoreSaved = false;
+
+            if (saveAsync && saveBinary)
             {
-                StringBuffer tmpStoreName;
-                constexpr const char *temporaryBinaryFileSaveName = "dali_store_tmp.bin";
-                OwnedIFileIO iBinaryFileIOTmpStore = createUniqueFile(location, temporaryBinaryFileSaveName, NULL, tmpStoreName);
-                iBinaryFileTmpStore.setown(createIFile(tmpStoreName));
-                try
+                // Execute saves in parallel only if both saves are enabled
+                Owned<CCompletionTask> completed = new CCompletionTask();
+
+                completed->spawn([this, &xmlCrc, &root, &iXmlFileIOTmpStore]() {
+                    LOG(MCdebugProgress, "Saving store async to XML");
+                    xmlCrc = saveStoreToFile(root, iXmlFileIOTmpStore, StoreFormat::XML);
+                });
+
+                if (saveBinary)
                 {
-                    binaryCrc = saveStoreToFile(root, iBinaryFileIOTmpStore, StoreFormat::BINARY);
-                    binaryCrcPtr = &binaryCrc;
+                    try
+                    {
+                        completed->spawn([this, &binaryCrc, &root, &iBinaryFileIOTmpStore, &binaryStoreSaved]() {
+                            LOG(MCdebugProgress, "Saving store async to binary");
+                            binaryCrc = saveStoreToFile(root, iBinaryFileIOTmpStore, StoreFormat::BINARY);
+                            binaryStoreSaved = true;
+                        });
+                    }
+                    catch (IException *e)
+                    {
+                        OWARNLOG(e, "Exception - Error saving async binary store file");
+                        e->Release();
+                        // If the binary save fails we ignore it and continue to allow the XML save to complete
+                    }
+                    catch (DALI_CATCHALL)
+                    {
+                        OWARNLOG("Unknown exception - Error saving async binary store file");
+                        // If the binary save fails we ignore it and continue to allow the XML save to complete
+                    }
                 }
-                catch (IException *e)
+
+                // Wait for all saves to complete
+                completed->decAndWait();
+            }
+            else
```

Comment:
This can be commoned up.
I suggest introducing a generic 'safeSaveStoreToFile', which handles the exception handling.
Also I think you can common up more, including the prefix creation, and the IFileIO (into saveStoreToFile), with it return the IFile (and crc) if it succeeds.